tests/conversion/utils_test.py

```diff
@@ -0,0 +1,37 @@
+import os
+import shutil
+import tempfile
+
+import pytest
+
+from osmaxx.conversion.converters.utils import recursive_getsize
+
+
+@pytest.fixture
+def directory_with_subdirectories(request):
+    tempdir = tempfile.TemporaryDirectory()
+
+    def rm_tmpdir():
+        shutil.rmtree(tempdir.name)
+    request.addfinalizer(rm_tmpdir)
+
+    file_sizes = [102, 304, 10, 2000, 10, 678691, 1667, 1254]
+    expected_size = sum(file_sizes)
+    subdir = tempdir.name
+    for index, file_size in enumerate(file_sizes):
+        if index % 2 == 0:
+            subdir = os.path.join(subdir, str(file_size))
```

Comment:
That's just a lot of code doing not helping to reduce repetitive work. What is the problem that you find hard to comprehend?